packages/backend/src/document.rs

```diff
@@ -349,3 +374,59 @@ pub async fn search_ref_stubs(
         items,
     })
 }
+
+pub async fn get_ref_children_stubs(ctx: AppCtx, ref_id: Uuid) -> Result<Vec<RefStub>, AppError> {
+    let user_id = ctx.user.as_ref().map(|u| u.user_id.clone());
+
+    let stub_rows = sqlx::query!(
+        r#"
+        WITH child_refs AS (
+            SELECT ARRAY_AGG(refs.id) AS child_ids
+            FROM refs
+            JOIN snapshots ON snapshots.id = refs.head
+            WHERE (
+                get_max_permission($2, refs.id) >= 'read'
+                AND jsonb_path_exists(
+                    snapshots.content,
+                    '$.*[*] ? (@._id == $id && (@.type == "diagram-in" || @.type == "analysis-of"))',
+                    jsonb_build_object('id', $1::uuid)
+                )
+            )
+        )
+        SELECT
+            stubs.ref_id           AS "ref_id!",
+            stubs.name,
+            stubs.type_name,
+            stubs.created_at       AS "created_at!",
+            stubs.permission_level AS "permission_level!: PermissionLevel",
+            stubs.owner_id,
+            stubs.owner_username,
+            stubs.owner_display_name
+        FROM
+            child_refs,
+            get_ref_stubs($2, child_refs.child_ids) AS stubs
```

Comment:
Currently this use of `get_ref_stubs` is overkill since we're only using `ref_id` on the client. The original idea was to ensure that whenever the server was returning a minimal reference to a ref it would return a `RefStub` to avoid a proliferation of utility types containing a subset of the same information. I'm no longer sure this is worth doing.

Reply:
Thanks for pointing this out. From the point of view of efficiency, it seems a small matter to return `RefStub`s instead of just ref IDs. Is the issue whether it's worth having a stored function `get_ref_stubs` if we only actually need it in one query, at least right now? I don't know. I leave it to your judgment.